File: src/main/Java/boj_step5/boj10809.java
import java.io.*;
import java.util.Arrays;

public class boj10809 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int[] alphabets = new int[26];
        Arrays.fill(alphabets, -1);
        char[] line = reader.readLine().toCharArray();
        int index = line.length;
        while (index-- > 0) alphabets[line[index] - 'a'] = index;
        for (int it: alphabets) writer.append(it + " ");
        writer.close();
    }
}

File: src/main/kotlin/boj10809.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val s = r.readLine()!!
    var alphabets = IntArray(26) { -1 }
    for (i in 0 until s.length) {
        if(alphabets[s.get(i) - 'a'] != -1) continue
        alphabets[s.get(i) - 'a'] = i
    }
    for (i in 0 until alphabets.size) w.write("${alphabets[i]} ")
    w.close()
}